cdp/wallet.py

```diff
@@ -384,6 +386,41 @@ def trade(self, amount: Number | Decimal | str, from_asset_id: str, to_asset_id:
 
         return self.default_address.trade(amount, from_asset_id, to_asset_id)
 
+    def invoke_contract(
+        self,
+        contract_address: str,
+        method: str,
+        abi: builtins.list[dict] | None = None,
+        args: dict | None = None,
+        amount: Number | Decimal | str | None = None,
+        asset_id: str | None = None,
+    ) -> ContractInvocation:
+        """Invoke a method on the specified contract address, with the given ABI and arguments.
+
+        Args:
+            contract_address (str): The address of the contract to invoke.
+            method (str): The name of the method to call on the contract.
+            abi (Optional[list[dict]]): The ABI of the contract, if provided.
+            args (Optional[dict]): The arguments to pass to the method.
+            amount (Optional[Union[Number, Decimal, str]]): The amount to send with the invocation, if applicable.
+            asset_id (Optional[str]): The asset ID associated with the amount, if applicable.
+
+        Returns:
+            ContractInvocation: The contract invocation object.
+
+        Raises:
+            ValueError: If the default address does not exist.
+
+        """
+        if self.default_address is None:
```

Comment:
slight different but in the typescript SDK we error if there is no default_address in the `defaultAddress()` function so this if check is not needed across all the functions. should we do that same pattern here?

Reply:
That's reasonable. I will take a pass through in a follow-up and key all of these off the can sign property.